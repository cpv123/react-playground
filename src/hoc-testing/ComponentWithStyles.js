import React from 'react';
import { 
    Button,
    ExpansionPanel,
    ExpansionPanelSummary,
    ExpansionPanelDetails,
    Typography,
    withStyles
} from '@material-ui/core';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';

const styles = theme => ({
    root: {
        width: '100%',
    },
    heading: {
        fontSize: theme.typography.pxToRem(15),
        fontWeight: theme.typography.fontWeightRegular,
    },
    button: {
        margin: theme.spacing.unit, 
    },
});

export function Component({ classes, children }) {
    return (
        <div className={classes.root}>
            <ExpansionPanel>

                <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
                    <Typography className={classes.heading}>
                        Expansion Panel Title
                    </Typography>
                </ExpansionPanelSummary>

                <ExpansionPanelDetails>
                    {children}
                    <Button 
                        className={classes.button}
                        color="primary"
                        variant="contained"
                        onClick={() => console.log('You clicked a well styled button')}
                    >
                        Click Me
                    </Button>
                </ExpansionPanelDetails>

            </ExpansionPanel>
        </div>
    );
}

const ComponentWithStyles = withStyles(styles)(Component);

export default ComponentWithStyles;
