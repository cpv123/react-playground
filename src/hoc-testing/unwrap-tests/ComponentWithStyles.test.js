import React from 'react';
import { shallow } from 'enzyme';
import { unwrap } from '@material-ui/core/test-utils';
import ComponentWithStyles from '../ComponentWithStyles';

const UnstyledComponent = unwrap(ComponentWithStyles);

describe('The ComponentWithStyles component', () => {

    it('renders as expected', () => {
        // Mock the classes prop from withStyles, otherwise test cannot run
        const wrapper = shallow(<UnstyledComponent classes={{}} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders children passed in', () => {
        const wrapper = shallow(
            <UnstyledComponent classes={{}}>
                <span>Click the button below!</span>
            </UnstyledComponent>
        );
        expect(wrapper.find('span')).toHaveLength(1);
    });

});