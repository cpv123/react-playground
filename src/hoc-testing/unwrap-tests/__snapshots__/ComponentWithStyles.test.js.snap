// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ComponentWithStyles component renders as expected 1`] = `
<div>
  <WithStyles(ExpansionPanel)>
    <WithStyles(ExpansionPanelSummary)
      expandIcon={<pure(ExpandMoreIcon) />}
    >
      <WithStyles(Typography)>
        Expansion Panel Title
      </WithStyles(Typography)>
    </WithStyles(ExpansionPanelSummary)>
    <WithStyles(ExpansionPanelDetails)>
      <WithStyles(Button)
        color="primary"
        onClick={[Function]}
        variant="contained"
      >
        Click Me
      </WithStyles(Button)>
    </WithStyles(ExpansionPanelDetails)>
  </WithStyles(ExpansionPanel)>
</div>
`;
